### PR TITLE
added scons ccache tool

### PR DESCRIPTION
**Pull request by dmoody256**
small port from mongodb tools, very useful.

I saw some mention of ccache on certain platforms detect, but it didn't look like it was used for the scons general build, and for sure not linuxbsd.

### CODE REVIEW

**Comment by dmoody256** [comment]
Also want to mention that this is more important with ninja generation tool. It is less useful with pure scons because scons has scons cache, and the ninja tool has not yet been updated to use the scons cache, so this ccache tool can work with ninja to provide build caching for the compilation at least, which is generally the bulk of build time.

**Comment by fire** [comment]
I have noticed large increase in performance on Linux systems. Can be salvaged independently from the Ninja pr.

File: misc/scons/site_tools/ccache.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+def exists(env):
+    """Look for a viable ccache implementation that meets our version requirements."""
+    if not env.subst("$ccache_bin"):
+        return False
+
+    ccache = env.WhereIs("$ccache_bin")
+
+    if not ccache:
+        print(f"Error: ccache not found at {env['ccache_bin']}")
```
[the remark this answers is not in the document]
Comment (dmoody256) [reply misc/scons/site_tools/ccache.py:43]: so this is fine as is?

**Comment by dmoody256** [comment]
> I have noticed large increase in performance on Linux systems. Can be salvaged independently from the Ninja pr.

This should work fine without ninja, but why not just use scons-cache if no ninja?

**Comment by akien-mga** [comment]
I guess #60698 is the most recent take on this if someone wants to salvage it, so we can close this one indeed.

**Pull request closed without merging**